tests/unit/features/DotFeatures_unittest.cc

```diff
@@ -121,3 +121,47 @@ TEST_F(DotFeaturesTest, compute_cov_nocopy)
 	for (index_t i = 0; i < (index_t)cov.size(); ++i)
 		EXPECT_NEAR(cov[i], ref_cov_ab[i], eps);
 }
+
+TEST_F(DotFeaturesTest, dense_dot_range)
+{
```

Comment:
Thanks for including the test!
What about including also a much more trivial one, whose expected result could be seen just from reading the test code in a second?
Also, I think it would be good to ensure test coverage with different number of threads. Is that currently happening?

Reply:
I believe it should run concurrently on any machine with `openmp`

Reply:
Let me rephrase, is there some test exercising this code with more than one thread?
Alternatively, have you tested it locally with different number of threads (at least one case with more than one).

Reply:
Locally `openmp` spawns 4 threads on my machine to do the operation.